src/key.rs

```diff
@@ -246,17 +246,34 @@ fn read_key(key: *mut raw::RedisModuleKey) -> Result<String, Utf8Error> {
     )
 }
 
-fn hash_get_key(
-    ctx: *mut raw::RedisModuleCtx,
-    key: *mut raw::RedisModuleKey,
-    field: &str,
-) -> Option<RedisString> {
-    let res = raw::hash_get(key, field);
-    if res.is_null() {
-        None
-    } else {
-        Some(RedisString::new(ctx, res))
+/// Get an arbitrary number of hash fields from a key by batching calls
+/// to `raw::hash_get_multi`.
+pub fn hash_mget_key<T>(ctx: *mut raw::RedisModuleCtx,
+                        key: *mut raw::RedisModuleKey,
+                        fields: &[T])
+    -> Result<Vec<Option<RedisString>>, RedisError>
+    where T: Into<Vec<u8>> + Clone
```

Comment:
I'm planning to look a bit deeper to understand the allocations and see if there are inefficiencies that can be avoided.